src/app/modules/main/node_section/view.nim

```diff
@@ -0,0 +1,188 @@
+import NimQml, chronicles, json, strutils
+import io_interface
+import ../../../core/signals/types
+import status/[fleet]
+
+
+
+QtObject:
+  type
+    View* = ref object of QObject
+      delegate: io_interface.AccessInterface
+      callResult: string
+      lastMessage*: string
+      wakuBloomFilterMode*: bool
+      fullNode*: bool
+      stats*: Stats
+      peerSize: int
+      bloomBitsSet: int
+      wakuV2LightClient: bool
+ 
+  proc delete*(self: View) =
+    self.QObject.delete
+
+  proc newView*(delegate: io_interface.AccessInterface): View =
+    new(result, delete)
+    result.QObject.setup
+    result.delegate = delegate
+    result.callResult = "Use this tool to call JSONRPC methods"
+    result.lastMessage = ""
+    result.wakuBloomFilterMode = false
+    result.fullNode = false
+    result.wakuV2LightClient = false
+
+  proc load*(self: View) =
+    self.delegate.viewDidLoad()
+
+  proc callResult*(self: View): string {.slot.} =
+    result = self.callResult
+
+  proc callResultChanged*(self: View, callResult: string) {.signal.}
+
+  proc setCallResult(self: View, callResult: string) {.slot.} =
+    if self.callResult == callResult: return
+    self.callResult = callResult
+    self.callResultChanged(callResult)
+
+  proc `callResult=`*(self: View, callResult: string) = self.setCallResult(callResult)
+
+  QtProperty[string] callResult:
+    read = callResult
+    write = setCallResult
+    notify = callResultChanged
+
+  proc onSend*(self: View, inputJSON: string) {.slot.} =
+    self.setCallResult(self.delegate.sendRPCMessageRaw(inputJSON))
+    echo "Done!: ", self.callResult
+
+  proc onMessage*(self: View, message: string) {.slot.} =
+    self.setCallResult(message)
+    echo "Received message: ", message
+
+  proc receivedMessage*(self: View, lastMessage: string) {.signal.}
+
+  proc setLastMessage*(self: View, lastMessage: string) {.slot.} =
+    self.lastMessage = lastMessage
+    self.receivedMessage(lastMessage)
+
+  QtProperty[string] lastMessage:
+    read = lastMessage
+    write = setLastMessage
+    notify = receivedMessage
+
+  proc initialized(self: View) {.signal.}
+
+  proc getWakuBloomFilterMode*(self: View): bool {.slot.} =
+    result = self.wakuBloomFilterMode
+
+  proc getBloomLevel*(self: View): string {.slot.} =
+    if self.wakuBloomFilterMode and not self.fullNode:
+      return "normal"
+    if self.wakuBloomFilterMode and self.fullNode:
+      return "full"
+    return "light"
+
+  QtProperty[bool] wakuBloomFilterMode:
+    read = getWakuBloomFilterMode
+    notify = initialized
+
+  QtProperty[string] bloomLevel:
+    read = getBloomLevel
+    notify = initialized
+
+  proc setWakuBloomFilterMode*(self: View, bloomFilterMode: bool) {.slot.} =
+    self.delegate.setBloomFilterMode(bloomFilterMode)
+    quit(QuitSuccess) # quits the app TODO: change this to logout instead when supported
+
+  proc init*(self: View) {.slot.} =
```

Comment:
@borismelnik not sure that I understand purpose of this proc and where is it called from?